lib/mix/lib/mix/task.ex

```diff
@@ -445,32 +499,60 @@ defmodule Mix.Task do
     end
   end
 
-  defp run_alias([h | t], alias_args, proj, original_task, _res) when is_binary(h) do
+  defp run_alias([h | t], alias_args, proj, original_task, options, _res) when is_binary(h) do
     case OptionParser.split(h) do
       [^original_task | args] ->
-        res = run_task(proj, original_task, args ++ alias_args)
-        run_alias(t, [], proj, original_task, res)
+        res = run_task(proj, original_task, args ++ alias_args, options)
+        run_alias(t, [], proj, original_task, options, res)
 
       [task | args] ->
-        res = run(task, join_args(args, alias_args, t))
-        run_alias(t, alias_args, proj, original_task, res)
+        res =
+          case options do
+            [apps: apps] ->
+              run_in_apps(task, apps, join_args(args, alias_args, t))
+
+            _ ->
+              run(task, join_args(args, alias_args, t))
+          end
```

Comment:
If you make `options` be a single `in_apps` variable that is either `nil` or a list of apps, you can rewrite this to:

    do_run(task, join_args(args, alias_args, t), in_apps)